### PR TITLE
Feat/nft amplitude

### PR DESCRIPTION
**Pull request by sunzsolo**
Adding Amplitude to NFTs

### CODE REVIEW

File: src/lib/components/nft/NftCard.tsx [hunk 1]
```diff
@@ -26,7 +27,12 @@ export const NftCard = ({
 
   return (
     <Flex direction="column" minW="full">
-      <AppLink href={`/nft-collections/${collectionAddress}/nft/${nftAddress}`}>
+      <AppLink
+        href={`/nft-collections/${collectionAddress}/nft/${nftAddress}`}
+        onClick={() => {
+          track(AmpEvent.USE_NFT_CARD, { showCollection });
+        }}
```
Comment (songwongtp) [inline src/lib/components/nft/NftCard.tsx:34]: () => track(AmpEvent.USE_NFT_CARD, { showCollection });

File: src/lib/pages/nft-collection-details/CollectionDetailsBody.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
+import {
```
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/CollectionDetailsBody.tsx:1]: why do we need this file?

File: src/lib/pages/nft-collection-details/index.tsx [hunk 3]
```diff
@@ -42,11 +42,11 @@ import { useCollectionInfos } from "./data";
 import type { CollectionDetailQueryParams } from "./types";
 import { TabIndex, zCollectionDetailQueryParams } from "./types";
 
-const InvalidCollection = () => (
+export const InvalidCollection = () => (
```
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/index.tsx:45]: remove export

File: src/lib/pages/nft-collection-details/index.tsx [hunk 4]
```diff
@@ -42,11 +42,11 @@ import { useCollectionInfos } from "./data";
...
   <InvalidState title="Collection does not exist" />
 );
 
-const tabHeaderId = "collectionDetailTab";
+export const tabHeaderId = "collectionDetailTab";
```
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/index.tsx:49]: ditto

File: src/lib/pages/nft-collections/components/Collections.tsx [hunk 5]
```diff
@@ -35,6 +37,11 @@ export const Collections = () => {
     }
   );
 
+  useEffect(() => {
+    if (router.isReady) track(AmpEvent.TO_NFT_COLLECTIONS_LIST);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router.isReady]);
+
   return (
     <Box>
```
Comment (songwongtp) [inline src/lib/pages/nft-collections/components/Collections.tsx:46]: change to fragment `<>`

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/nft-collection-details/index.tsx (hunk 4 above)
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/index.tsx:49]: line 110 change to fragment `<>`

File: src/lib/pages/nft-collection-details/components/tables/activities/index.tsx [hunk 6]
```diff
@@ -56,6 +56,7 @@ export const Activities = ({
           setCurrentPage(1);
         }}
         size={{ base: "md", md: "lg" }}
+        amptrackSection="activities-in-this-collection-search"
```
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/components/tables/activities/index.tsx:59]: line 47, discuss with @jennieramida

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/nft-collection-details/CollectionDetailsBody.tsx (hunk 2 above)
Comment (sunzsolo) [reply src/lib/pages/nft-collection-details/CollectionDetailsBody.tsx:1]: Deleted

File: src/lib/pages/nft-collection-details/index.tsx (hunk 3 above)
Comment (sunzsolo) [reply src/lib/pages/nft-collection-details/index.tsx:45]: fixed

File: src/lib/pages/nft-collection-details/index.tsx (hunk 4 above)
Comment (sunzsolo) [reply src/lib/pages/nft-collection-details/index.tsx:49]: ditto

File: src/lib/pages/nft-collections/components/Collections.tsx (hunk 5 above)
Comment (sunzsolo) [reply src/lib/pages/nft-collections/components/Collections.tsx:46]: fixed

File: src/lib/pages/nft-collection-details/index.tsx (hunk 4 above)
Comment (sunzsolo) [reply src/lib/pages/nft-collection-details/index.tsx:49]: fixed

File: src/lib/pages/nft-collection-details/components/tables/activities/index.tsx (hunk 6 above)
Comment (sunzsolo) [reply src/lib/pages/nft-collection-details/components/tables/activities/index.tsx:59]: fixed

File: src/lib/components/nft/NftCard.tsx (hunk 1 above)
Comment (sunzsolo) [reply src/lib/components/nft/NftCard.tsx:34]: fixed

File: src/lib/pages/nft-collection-details/components/tables/activities/ActivitiesTableMobileCard.tsx [hunk 7]
```diff
@@ -39,7 +39,12 @@ export const ActivitiesTableMobileCard = ({
       }
       topContent={
         <Flex align="center" gap={2}>
```
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/components/tables/activities/ActivitiesTableMobileCard.tsx:41]: try removing this

File: src/lib/pages/nft-collection-details/index.tsx [hunk 8]
```diff
@@ -166,6 +166,11 @@ const CollectionDetailsBody = ({
             w={{ base: "full", md: "auto" }}
             size={{ base: "sm", md: "md" }}
             mb={{ base: 4, md: 0 }}
+            onClick={() => {
```
Comment (songwongtp) [inline src/lib/pages/nft-collection-details/index.tsx:169]: `() => track(AmpEvent.USE_NFT_VIEW_RESOURCE_CTA, {
                amptrackSection: "nft-collection-details",
              })`

Comment (sunzsolo) [reply src/lib/pages/nft-collection-details/index.tsx:169]: fixed

**Review by songwongtp — APPROVED** [review]

File: src/lib/pages/nft-collection-details/data.ts [hunk 9]
```diff
@@ -20,39 +19,34 @@ export const useCollectionInfos = (
...
+    .find((group) => group.group === "collection")
+    ?.items.find(
+      (resource) =>
+        resource.structTag === "0x1::collection::FixedSupply" ||
+        resource.structTag === "0x1::collection::UnlimitedSupply"
+    );
+
+  const isUnlimited =
+    collectionSupplyResource?.structTag === "0x1::collection::UnlimitedSupply";
+  const supplyData = collectionSupplyResource
```
Comment (evilpeach) [inline src/lib/pages/nft-collection-details/data.ts:37]: I think we should declare type for `supplyData`.

File: src/lib/pages/nft-collection-details/data.ts [hunk 10]
```diff
@@ -20,39 +19,34 @@ export const useCollectionInfos = (
...
-    })
-    .map((data) => data?.data);
-
-  const [type] = collectionSupplies;
-  const isUnlimited = type?.structTag === "0x1::collection::UnlimitedSupply";
-
-  const [supplyData] = parsed;
-  const supplies = snakeToCamel(supplyData);
+  const collectionRoyalty = royalty
+    ? JSON.parse(royalty.items[0].moveResource)
```
Comment (evilpeach) [inline src/lib/pages/nft-collection-details/data.ts:24]: If the length of items is zero, could that potentially cause a problem?

**Review by evilpeach — APPROVED** [review]

**Pull request merged**